pkg/operator/resources/operator/operator_test.go

```diff
@@ -0,0 +1,62 @@
+package operator
+
+import (
+	"fmt"
+	"github.com/RHsyseng/operator-utils/pkg/validation"
+	"github.com/ghodss/yaml"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	cdiv1alpha1 "kubevirt.io/containerized-data-importer/pkg/apis/core/v1alpha1"
+
+	"strings"
+	"testing"
+)
+
+var crdTypeMap = map[string]interface{}{
+	OperatorCdiCRD: &cdiv1alpha1.CDI{},
+}
+
+func TestCRDSchemas(t *testing.T) {
+	for crdFileName, cdiObjType := range crdTypeMap {
+
+		schema := getSchema(t)
+		missingEntries := schema.GetMissingEntries(cdiObjType)
+		for _, missing := range missingEntries {
+			if strings.HasPrefix(missing.Path, "/status") {
+				//Not using subresources, so status is not expected to appear in CRD
+			} else {
+				assert.Fail(t, "Discrepancy between CRD and Struct",
+					"Missing or incorrect schema validation at %v, expected type %v  in CRD file %v", missing.Path, missing.Type, crdFileName)
+			}
+		}
+	}
+}
+
+func TestSampleCustomResources(t *testing.T) {
+
+	var crFileName, crdFileName string = "cdi-cr.yaml", OperatorCdiCRD
```

Comment:
How about some negative tests?  And rather than storing a bunch of manifests in files, how about creating a bunch of CDIs in code and serializing them to yaml?

Reply:
Oh yeah CDIs in code prob won’t work for negative test cases would have to use something like k8s unstructured or raw strings

Reply:
> Oh yeah CDIs in code prob won’t work for negative test cases would have to use something like k8s unstructured or raw strings

added a negative test case for CR validation.